### ISSUE TITLE
False positive: Cannot declare a new module at this location

### ISSUE THREAD

**Issue by Songbird0** [intellij-rust/intellij-rust#1638]
Hi!

I've put my `pub mod generator;` statement in `lib.rs` file but, unlike cargo, Intellij has displayed an error which isn't one.

I tried this code:

```rust
// in `lib.rs` file

pub mod generator;
```

Here's my `tree`:

```
.
├── Cargo.lock
├── Cargo.toml
├── curator.iml
├── LICENSE
├── src
│   ├── generator   <------|
│   │   └── mod.rs          HERE
│   ├── lib.rs           <------|
│   └── main.rs
├── target
│   ├── debug
│   │   ├── build
│   │   ├── curator
│   │   ├── curator-50ba615edf19a313
│   │   ├── curator-50ba615edf19a313.d
│   │   ├── curator-c0da80fad1d5f504
│   │   ├── curator-c0da80fad1d5f504.d
│   │   ├── curator.d
│   │   ├── curator_tests-6f6a999f42e7aa6c
│   │   ├── curator_tests-6f6a999f42e7aa6c.d
│   │   ├── deps
│   │   │   ├── curator-09970411e20ec522
│   │   │   ├── curator-50ba615edf19a313
│   │   │   ├── curator-9ac71e677db89c15.d
│   │   │   ├── curator-c0da80fad1d5f504
│   │   │   ├── curator_tests-6f6a999f42e7aa6c
│   │   │   ├── libansi_term-76963246a5b3dac4.rlib
│   │   │   ├── libansi_term-f4a35a7944a9cc65.rmeta
│   │   │   ├── libatty-723ef0f722dadd40.rmeta
│   │   │   ├── libatty-77a28a90e81f9739.rlib
│   │   │   ├── libbitflags-3595fc3dc3bf92a4.rmeta
│   │   │   ├── libbitflags-6808a3e0b5724221.rlib
│   │   │   ├── libclap-51f0998050664b54.rlib
│   │   │   ├── libclap-c164a8714cb91a14.rmeta
│   │   │   ├── libcurator-95aef8f90edfbf1d.rlib
│   │   │   ├── liblibc-2236279eda551aca.rlib
│   │   │   ├── liblibc-e833de71e85d74b5.rmeta
│   │   │   ├── librand-08b94fc86a85f9cd.rlib
│   │   │   ├── librand-9ab2073a5d26a327.rmeta
│   │   │   ├── libstrsim-05d8f212bfa8ec8c.rmeta
│   │   │   ├── libstrsim-4ff5db0351b0fb02.rlib
│   │   │   ├── libterm_size-7480b3ba144eca68.rlib
│   │   │   ├── libterm_size-cd8d4afc4a48ff9b.rmeta
│   │   │   ├── libtextwrap-56722955018e17d0.rlib
│   │   │   ├── libtextwrap-75820c4200b8bb1b.rmeta
│   │   │   ├── libunicode_segmentation-4314d4f8ce7c7708.rmeta
│   │   │   ├── libunicode_segmentation-e7bc06523d0c2ccb.rlib
│   │   │   ├── libunicode_width-c043bf8889809421.rmeta
│   │   │   ├── libunicode_width-cff4ed7c5f0be904.rlib
│   │   │   ├── libvec_map-66a4aac55f7a1f7f.rlib
│   │   │   └── libvec_map-9ae28b94a743477c.rmeta
│   │   ├── examples
│   │   ├── incremental
│   │   ├── libcurator.d
│   │   ├── libcurator.rlib
│   │   └── native
│   └── release
│       ├── build
│       ├── deps
│       │   ├── libansi_term-44c95d2f92186b69.rlib
│       │   ├── libatty-9ec37e7a6430560c.rlib
│       │   ├── libbitflags-99b0cafaace6f782.rlib
│       │   ├── libclap-ca264b0a270a0526.rlib
│       │   ├── liblibc-0547e85bf76474ea.rlib
│       │   ├── librand-17d26da4017a015e.rlib
│       │   ├── libstrsim-09d9ddaa2d2a5eee.rlib
│       │   ├── libterm_size-06f832832405a316.rlib
│       │   ├── libtextwrap-98fb4ef7683bc6e7.rlib
│       │   ├── libunicode_segmentation-85bc3e2d05f59b68.rlib
│       │   ├── libunicode_width-0419a2e7d9b7a1fa.rlib
│       │   └── libvec_map-dcb0e1e17a4d7879.rlib
│       ├── examples
│       ├── incremental
│       └── native
└── tests
    └── curator_tests.rs

```

I expected to see this happen: `pub mod generator;` statement isn't underlined.

Instead, this happened: `pub mod generator;` statement is underlined. See below:

![image](https://user-images.githubusercontent.com/13397428/29474771-64b7a35c-845d-11e7-99bf-8987ae43073c.png)

`cargo check` result:

```rust
   Compiling curator v0.1.0 (file:///...)
warning: unused import: `std::result::Result`
  --> src/generator/mod.rs:25:5
   |
25 | use std::result::Result;
   |     ^^^^^^^^^^^^^^^^^^^
   |
   = note: #[warn(unused_imports)] on by default

warning: unused variable: `i`
  --> src/generator/mod.rs:99:13
   |
99 |         for i in 0..occcurrences {
   |             ^
   |
   = note: #[warn(unused_variables)] on by default

warning: unused variable: `occurrences`
   --> src/generator/mod.rs:111:31
    |
111 |     pub fn gen_all_pwd(&self, occurrences: usize, pwd_number: usize) -> String {
    |                               ^^^^^^^^^^^
    |
    = note: #[warn(unused_variables)] on by default

warning: unused variable: `pwd_number`
   --> src/generator/mod.rs:111:51
    |
111 |     pub fn gen_all_pwd(&self, occurrences: usize, pwd_number: usize) -> String {
    |                                                   ^^^^^^^^^^
    |
    = note: #[warn(unused_variables)] on by default

    Finished dev [unoptimized + debuginfo] target(s) in 0.31 secs
```
There are warnings but no errors.

## Meta

rustc -V: `rustc 1.19.0 (0ade33941 2017-07-17)`
cargo -V: `cargo 0.20.0 (a60d185c8 2017-07-13)`

Regards.

**Comment by kumbayo** [comment]
I could reproduce the error the following way:
git clone https://github.com/Songbird0/Curator
git checkout d8087dc6ee75edd5def5a16329932a26b57b0710
Open the project in IntelliJ
git checkout 939ee548e8084c83e28ee19c0e317daf7a5db8bb
-> The error appears at src/lib.rs

**Comment by Redrield** [comment]
Getting this same issue with the latest version of the plugin.

**Comment by lenovouser** [comment]
This issue still exists for my friend @phenomax who isn't able to comment at this time 😏 

```
windows:  10
intellij: 2017.2.6
plugin:   0.2.0.2079
rustc:    1.21.0 (3b72af97e 2017-10-09)
cargo:    0.22.0 (3423351a5 2017-10-06)
java:     1.8.0_152-release-915-b12 amd64
```

**Comment by kumbayo** [comment]
@lenovouser If this issue still exists in the current version can you please provide more details on how to reproduce the issue? Thanks

**Comment by lenovouser** [comment]
Ping @phenomax
